andela/ah-scorpion-frontend
# #160345747 Enable User Logout

#### What does this PR do?
- adds a function to handle user logout request event

#### Description of Task to be completed?
- added a logout function to implement logout on Redux and clear user data from localStorage
- added `history prop` to the `UserNavBar` to handle browser history
- added a `handleLogout` function to handle the request when the user clicks the logout link

#### How should this be manually tested?
- use git to pull for the branch `ft-user-logout-160345747`
- update your environmental variables on the `.env` file. Be sure to set a value for the `REACT_APP_API_URL` to point to the back-end server
- run `source .env` to export the env variables
- run `npm install` to update dependencies
- run `npm start -s` to start the server
- use the URL `/login' to log in a user
- once logged in copy the URL to the dashboard and paste in a different tab on the same browser window. The dashboard should still be accessible.
- on the right end of the blue navigation bar, click on the drop-down link next to the username then click `Sign Out`
- The URL should change to `/login` and you will be taken to the login window.
- If you try accessing the `/dashboard` URL you copied earlier, you will be taken back to the login window

**Note:** If you are using a local server for the backend and experience issues receiving responses, you might consider using a project hosted on `heroku`

#### What are the relevant pivotal tracker stories?
- [#160345747](https://www.pivotaltracker.com/story/show/160345747)
